temporian/implementation/numpy_cc/operators/custom_heap.h

```diff
@@ -0,0 +1,76 @@
+#include <iostream>
+#include <list>
+template <typename T>
+class CustomHeap {
+ private:
+  std::function<bool(T, T)> compare;
+  std::list<T> heap;
+  std::unordered_map<T, typename std::list<T>::iterator> val_to_node;
+
+ public:
+  CustomHeap(std::function<bool(T, T)> compare) : compare(compare) {}
+
+  void push(T value) {
+    heap.push_back(value);
```

Comment:
This is invalidating all the iterators in "val_to_node" right? Can you add a comment.

Reply:
It shouldn't because heap is a double-linked list so all the existing nodes should remain in the same place. That's the reason why I'm using lists instead of vectors